src/main/java/org/sagebionetworks/bridge/services/PermissionService.java

```diff
@@ -158,6 +172,120 @@ public void deletePermission(String appId, String guid) {
+        
+        Map<EntityType, Map<String, Set<AccessLevel>>> additionalPermissions = new HashMap<>();
+        Map<EntityType, Map<String, Set<AccessLevel>>> removablePermissions = new HashMap<>();
+        
+        for (EntityType entityType : newPermissions.keySet()) {
+            if (oldPermissions.containsKey(entityType)) {
+                // check which permissions are already accounted for, add any missing
+                Map<String, Set<AccessLevel>> newPermissionEntityTypeMap = newPermissions.get(entityType);
+                for (String entityId : newPermissionEntityTypeMap.keySet()) {
+                    if (oldPermissions.get(entityType).containsKey(entityId)) {
+                        Set<AccessLevel> oldAccessLevels = oldPermissions.get(entityType).get(entityId);
+                        Set<AccessLevel> additionalAccessLevels = new HashSet<>();
+                        for (AccessLevel accessLevel : newPermissionEntityTypeMap.get(entityId)) {
+                            if (!oldAccessLevels.contains(accessLevel)) {
+                                additionalAccessLevels.add(accessLevel);
+                            }
+                        }
+                        Map<String, Set<AccessLevel>> entityTypeMap = additionalPermissions.getOrDefault(entityType, new HashMap<>());
+                        entityTypeMap.put(entityId, additionalAccessLevels);
+                        additionalPermissions.put(entityType, entityTypeMap);
+                    } else {
+                        Map<String, Set<AccessLevel>> entityTypeMap = additionalPermissions.getOrDefault(entityType, new HashMap<>());
+                        entityTypeMap.put(entityId, newPermissionEntityTypeMap.get(entityId));
+                        additionalPermissions.put(entityType, entityTypeMap);
+                    }
+                }
+            } else {
+                // the old permissions do not have any of the new so add them all
+                additionalPermissions.put(entityType, newPermissions.get(entityType));
+            }
+        }
+        
+        for (EntityType entityType : oldPermissions.keySet()) {
+            if (newPermissions.containsKey(entityType)) {
+                Map<String, Set<AccessLevel>> oldPermissionEntityTypeMap = newPermissions.get(entityType);
+                for (String entityId : oldPermissionEntityTypeMap.keySet()) {
+                    if (newPermissions.get(entityType).containsKey(entityId)) {
+                        Set<AccessLevel> newAccessLevels = newPermissions.get(entityType).get(entityId);
+                        Set<AccessLevel> removableAccessLevels = new HashSet<>();
+                        for (AccessLevel accessLevel : oldPermissionEntityTypeMap.get(entityId)) {
+                            if (!newAccessLevels.contains(accessLevel)) {
+                                removableAccessLevels.add(accessLevel);
+                            }
+                        }
```

Comment:
We are missing unit tests where the new permission type has the same entity type and entity, but removes permissions from that entity (eg, an entity going from ListThroughDelete to ListRead, which removes Edit and Delete).

Reply:
Yep, adjusted the main delete test to go from ADMIN to ORG_ADMIN to cover this as well.